spirv_cross.cpp

```diff
@@ -4149,3 +4149,30 @@ bool Compiler::instruction_to_result_type(uint32_t &result_type, uint32_t &resul
 			return false;
 	}
 }
+
+void Compiler::rename_interface_variable(const std::vector<Resource> &resources,
```

Comment:
I'm almost inclined to think this should live in a separate file, e.g. spirv_cross_util.cpp. There is no particular reason why this needs to live in spirv_cross.cpp. It's also loaded a bit too much with "policy", not mechanics.

The arguments are not well suited for a public API. InterfaceVariableRename is an internal structure which species the storage class as well, but that's redundant, because it's implied by the `resources` argument. Something like `const std::vector<Resource> &resources, uint32_t location, const std::string &name)` would work much better.